src/main/python/systemds/operator/nodes/matrix.py

```diff
@@ -634,6 +634,14 @@ def to_string(self, **kwargs: Dict[str, VALID_INPUT_TYPES]) -> "Scalar":
             self.sds_context, "toString", [self], kwargs, output_type=OutputType.STRING
         )
 
+    def to_scalar(self):
+        return Scalar(self.sds_context, "as.scalar", [self])
+
+    def to_frame(self):
+        from systemds.operator import Frame
```

Comment:
cyclic dependency, i understand it is how i thought about it as well, however, maybe we need to add it afterwards dynamically instead.

See:

https://stackoverflow.com/questions/972/adding-a-method-to-an-existing-object-instance-in-python

This would allow us to add the to_scalar, to_frame and to_matrix on the operation_node directly inside frame, matrix and scalar themselves to overwrite the operation node.

This would also allow us to have documentation on the method, since you add an empty method with docs that would throw an exception on the operation_node, and then on the import of Matrix etc it overwrite the Class method such that new instances use the overwritten method in Matrix.

Reply:
I see, currently the cyclic dependency error is avoided since the import happens inside the function, which won't be executed while importing afaik.